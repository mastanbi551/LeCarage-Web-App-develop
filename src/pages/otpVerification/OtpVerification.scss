@import "../../styles/variables.scss";

.otp-verification-page {
  &__heading {
    padding: 20px 20px 20px 0;
  }
  &__content {
    color: $primaryWhite;
  }
  &__container {
    display: flex;
    flex-direction: row;
    @media (max-width: 991px) {
      flex-direction: column-reverse;
    }
  }
  &__logo-container {
    flex-basis: 20%;
    padding-top: 3%;
    @media (max-width: 991px) {
      padding-top: 5% !important;
    }
  }
  &__logo-img {
    /*to be updated.*/
    color: transparent;
  }

  &__otp-verification-form {
    padding: 50px;
    margin-left: 10%;
    flex-basis: 50%;
    @media (max-width: 991px) {
      width: 320px !important;
      margin: auto !important;
      padding: 50px 0px !important;
    }

    &__otp-field {
      padding: 0 20px 20px 0;
      color: $primaryWhite !important;
    }

    &__form-fields {
      display: flex;
      flex-direction: column;
      align-items: flex-start;
    }

    &__proceed-button {
      margin-top: 6%;
    }
  }

  &__checkbox-forgot {
    display: flex;
  }

  input:disabled {
    background: $secondaryLightGrey;
    cursor: not-allowed;
  }
  .static {
    border: 1px solid $lightgrey;
    background-color: transparent;
    box-sizing: border-box;
    width: 100%;
    height: 40px;
    min-width: 19.875rem;
  }
}
