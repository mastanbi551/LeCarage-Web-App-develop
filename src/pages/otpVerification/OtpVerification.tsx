import React from 'react';
import { Formik } from 'formik';
import {
  BackgroundImageContainer,
  InputTextField,
  PageHeader,
  LeCarageLogo,
  CustomButton,
  Loader,
} from '../../components';
import { ImagesPath, ConstantStrings } from '../../utils/constants';
import '../otpVerification/OtpVerification.scss';
import { OtpValidationSchema } from '../../utils/ValidationSchemas';

interface OtpVerifyProps {
  handleOnSubmit: (values: {otp: string})=> void;
  showLoader: boolean
}

export const OtpVerification:React.FC<OtpVerifyProps>  = (
  {handleOnSubmit, showLoader}
) => {
    
    return (
      
      <Formik
        initialValues={{otp: ''}}
        validationSchema={OtpValidationSchema}
        onSubmit={values => {
          handleOnSubmit(values);
        }}
      >
        {({handleChange, handleSubmit, values, errors, isValid, handleBlur, touched}) => (
            <BackgroundImageContainer imageSource={ImagesPath.bgImageWithCar}>
              <>
                  <div className = 'otp-verification-page'>
                    <div className='otp-verification-page__container'>
                        <div className='otp-verification-page__otp-verification-form'>
                            <div className='otp-verification-page__heading'>
                                <PageHeader text={ConstantStrings.otpVerificationPageHeading} name='heading'/>
                            </div>
                            <div className='otp-verification-page__content'>
                                <p className='body_copy_1'>{ConstantStrings.otpVerificationPage.content}</p>
                            </div>
                            <form onSubmit={handleSubmit} onKeyDown={(e) => {
                    if(e.key === 'Enter')
                    handleSubmit();}}className='otp-verification-page__otp-verification-form__form-fields'>
                                <div className="otp-verification-page__otp-verification-form__otp-field">
                                <InputTextField 
                                            name='input-field' 
                                            dataTestId='otp-input'
                                            id='otp_field' 
                                            placeholder={ConstantStrings.otpVerificationPlaceholder}
                                            value={values.otp} 
                                            inputType='password'
                                            requiredOrNot={true}
                                            onChangeText={handleChange('otp')}
                                            error={errors.otp}
                                            onBlur={handleBlur('otp')}
                                            touched={touched.otp}
                                >
                                  {undefined}
                                </InputTextField>
                                
                                </div>
                                
                              
                                <div className='otp-verification-page__otp-verification-form__proceed-button'>
                                        <CustomButton 
                                        dataTestId='submit-button'            
                                            title="Proceed"
                                            name={showLoader ? 'cursor-not-allowed button_full': 'button_full'} 
                                            buttonOnClick={handleSubmit}
                                            isDisable={!isValid}
                                        />
                                </div>                                       
                            </form>
                      </div>
                      {
                                showLoader && <Loader />
                            }
                      <div className='otp-verification-page__logo-container'>               
                        <LeCarageLogo source={ImagesPath.lecaragelogo} altText="logo" name='logo-page__img'/>                
                      </div>
                    </div>  
                    </div>  
              </>
                
            
            </BackgroundImageContainer>
          )
        }
      </Formik>
    );
};