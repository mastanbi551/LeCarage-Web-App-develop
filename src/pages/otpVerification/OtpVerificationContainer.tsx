import React, { useEffect, useState } from 'react';
import { useNavigate } from 'react-router';
import { useAppSelector } from '../../hooks/Hooks';
import { useDispatch } from 'react-redux';
import { onLoadingStop, OtpVerifyRequest } from '../otpVerification/otpState/OtpVerifySlice';
import { OtpVerification } from './OtpVerification';
import { responseStatus } from '../../utils/constants/ConstantStrings';

export const OtpVerificationContainer = () => {
    const { uId } = useAppSelector(state => state.forgotPasswordReducer);
    const [userId, setUserId] = useState(uId !== '' ? uId : localStorage.getItem('uId'));
    const navigate = useNavigate();
    const dispatch = useDispatch();
    const [showLoader, setShowLoader] = useState(false);
    const { responseData, isLoading } = useAppSelector(
        (state) => state.otpVerifyReducer
      );
    
      useEffect(() => {
        if (isLoading) {
          setShowLoader(false);
          if (responseData.status === responseStatus.success) {
            dispatch(onLoadingStop());
            navigate('/resetpassword');
          }
        }
      }, [isLoading, responseData]);

      const handleOnSubmit = (values: {otp: string}) => {
        setShowLoader(true);
        const params = JSON.stringify({
          userID: userId,
          otpToVerify: values.otp
        });
        dispatch(OtpVerifyRequest(params));
      };  
   
      return <OtpVerification 
        handleOnSubmit={handleOnSubmit} 
        showLoader={showLoader}
      />;


};