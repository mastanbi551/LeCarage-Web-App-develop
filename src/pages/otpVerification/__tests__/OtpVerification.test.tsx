import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { OtpVerification } from '../../otpVerification/OtpVerification';

describe('OtpVerification component', () => {
  const handleOnSubmit = jest.fn();

  it('matches the snapshot', () => {
    const { container } = render(<OtpVerification handleOnSubmit={handleOnSubmit} />);
    expect(container).toMatchSnapshot();
  });

  it('submits the form with the entered OTP', () => {
    const { getByTestId } = render(<OtpVerification handleOnSubmit={handleOnSubmit} />);
    const otpInput = getByTestId('otp-input');
    const submitButton = getByTestId('submit-button');
  
    fireEvent.change(otpInput, { target: { value: '123456' } });
    fireEvent.click(submitButton);
  
   
    expect(handleOnSubmit).toHaveBeenCalled;
  });
  
});
