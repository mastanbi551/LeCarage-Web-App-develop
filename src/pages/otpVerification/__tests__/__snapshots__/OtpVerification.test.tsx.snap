// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OtpVerification component matches the snapshot 1`] = `
<div>
  <div
    data-testid="background-image-container"
    style="background-image: url(bgimg.jpg); background-size: cover; background-repeat: no-repeat; background-position: center; height: 100vh; overflow: hidden;"
  >
    <div
      class="otp-verification-page"
    >
      <div
        class="otp-verification-page__container"
      >
        <div
          class="otp-verification-page__otp-verification-form"
        >
          <div
            class="otp-verification-page__heading"
          >
            <h1
              class="heading"
            >
              Enter Your OTP
            </h1>
          </div>
          <div
            class="otp-verification-page__content"
          >
            <p
              class="body_copy_1"
            >
              Please enter your verification code 
            </p>
          </div>
          <form
            class="otp-verification-page__otp-verification-form__form-fields"
          >
            <div
              class="otp-verification-page__otp-verification-form__otp-field"
            >
              <div
                class="input-text-field"
              >
                <input
                  autocomplete="new-password"
                  autocorrect="off"
                  class="input-field static"
                  data-testid="otp-input"
                  id="otp_field"
                  placeholder="Enter 6-digit OTP"
                  required=""
                  type="password"
                  value=""
                />
              </div>
            </div>
            <div
              class="otp-verification-page__otp-verification-form__proceed-button"
            >
              <button
                class="button_full"
                data-testid="submit-button"
                type="button"
              >
                Proceed
              </button>
            </div>
          </form>
        </div>
        <div
          class="otp-verification-page__logo-container"
        >
          <img
            alt="logo"
            class="logo-page__img"
            src="lecarageLogo.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
